### PR TITLE
Demon bloodcrawl balance

### PR DESCRIPTION
**Pull request by Xackii**
## About The Pull Request

Demon now takes damage when located in bloodcrawl(every 20 seconds take 5% of max demon hp).
    When demon first appears, it does not take damage while in Jaunt. He also doesn't take damage while he's eating someone.
Demon now can't eat simple animals/basic mobs and etc. Only carbon human being.

## Why It's Good For The Game

Makes demons less immortal.

## Changelog

:cl:
balance: Demon: bloodcrawl now deal damage when you using it.
balance: Demon: demon can eat only carbon human beings.
/:cl:

### CODE REVIEW

**Comment by improvedname** [comment]
I see the following issues with this PR

While demons are a plague amongst the crew i do not want you to forget that this antag is bought by a wizard who can practically touch you want a hand to instantly kill you and round remove you.

20 seconds timer is too short in my opinion a demon will not always get the oppurtunity to do their thing and the 5% damage each 20 seconds will only make them only play them even safer. i would either suggest a damage cap so it would stop at 50% health so the demon would be left with 150 HP so they'd die on 6 laser dies(which is still super reasonable)

another issue with this is the reward you get from killing a blood demon practically damaging you every 20 seconds players will still take kpa damage and cold/heat im bloodcrawl and that ontop of the bloodcrawl taking 5% of your health every 20 seconds is kinda ridicilous 

Another issue with demons is you dont know where they are coming from unless you are paying attenion to your chatbox it would be very reasonable to make a balloon alert you can see where blood is bubbling which kinda ruins the suprise of the demon getting out but atleast you'd be able to prepare against it

**Comment by zoomachina** [comment]
> While demons are a plague amongst the crew i do not want you to forget that this antag is bought by a wizard who can practically touch you want a hand to instantly kill you and round remove you.

Currently, demons are oftentimes deadlier than wizards themselves, despite the fact that wizards have the smite spell, like you mentioned
> 20 seconds timer is too short in my opinion a demon will not always get the oppurtunity to do their thing and the 5% damage each 20 seconds will only make them only play them even safer. i would either suggest a damage cap so it would stop at 50% health so the demon would be left with 150 HP so they'd die on 6 laser dies(which is still super reasonable)

It's enough to get around the station. I guess the demon won't have much time to chill, but a certain other pr got closed with the reason "we don't want traitors to chill", so..
> Another issue with demons is you dont know where they are coming from unless you are paying attenion to your chatbox it would be very reasonable to make a balloon alert you can see where blood is bubbling which kinda ruins the suprise of the demon getting out but atleast you'd be able to prepare against it

That's true, but it's not a problem with this pr

**Comment by DarkHunter7756** [comment]
> wizard who can practically touch you want a hand to instantly kill you and round remove you

Smite Spell costs 2 points and without upgrade needs 60 seconds to wait. Im not sure, but maybe even brain is falling out. Demon can eat without recharging, there can be 2 demons at same cost. You cant just say "wizard not about balance" because he has balance 

> 20 seconds timer is too short in my opinion

You not take timer damage outside Jaunt. 

> demon would be left with 150 HP so they'd die on 6 laser dies

Usually you have 1-5 seconds before he disappear. And no lasers. Revenant experience for example. Demon never attack security. 

> players will still take kpa damage and cold/heat

Well noted. It's strange that he even takes damage from these hazards

**Comment by improvedname** [comment]
Its better to limit the amount wizard can buy.

Add a blood crawl balloon alert where they are jaunting from and that would already adress the main issue of demon spam



> > wizard who can practically touch you want a hand to instantly kill you and round remove you
> 
> Smite Spell costs 2 points and without upgrade needs 60 seconds to wait. Im not sure, but maybe even brain is falling out. Demon can eat without recharging, there can be 2 demons at same cost. You cant just say "wizard not about balance" because he has balance

Slaughter demons the round removal ones cost 2 points where laughter costs 1

**Comment by DarkHunter7756** [comment]
> Slaughter demons the round removal ones cost 2 points where laughter costs 1

They all round removal. Its not important dead you will be or alive, because you never escape him

**Comment by Xackii** [comment]
> While demons are a plague amongst the crew i do not want you to forget that this antag is bought by a wizard who can practically touch you want a hand to instantly kill you and round remove you.
>
The fact of the matter is that this is the call of a wizard and not a separate antagonist. As I said in past PR, if it were a separate antagonist, then everything would be fine. But as for me, demon should not live long and its benefits for wizard should be visible immediately, otherwise I see no other reasons to buy a demon (except to ruin the round)
> 20 seconds timer is too short in my opinion a demon will not always get the oppurtunity to do their thing and the 5% damage each 20 seconds will only make them only play them even safer. i would either suggest a damage cap so it would stop at 50% health so the demon would be left with 150 HP so they'd die on 6 laser dies(which is still super reasonable)
> 
In such cases, nothing will change. If demon doesn't do anything then let him die
> another issue with this is the reward you get from killing a blood demon practically damaging you every 20 seconds players will still take kpa damage and cold/heat im bloodcrawl and that ontop of the bloodcrawl taking 5% of your health every 20 seconds is kinda ridicilous
> 
Unlike a demon, the one who received the heart can simply heal himself or inject a reagent that will heal him, so I don’t see a problem with this.

**Comment by CRITAWAKETS** [comment]
Slaughter Demon is currently super weak as it is, having relatively low melee-only damage for a lack of speed aside from ambushing and an incredibly easy weakness (just pull out space cleaner).

I've only seen slaughter demons somewhat recently do nothing but fail from experience because they can't hit shit whilst being countered by incredibly easily distributable cleaning supplies. If you're giving them nerfs, atleast give some sort of thing to make it so that they aren't countered by as something as simple as researching advanced mops.

**Comment by zoomachina** [comment]
> I've only seen slaughter demons somewhat recently do nothing but fail from experience because they can't hit shit whilst being countered by incredibly easily distributable cleaning supplies. If you're giving them nerfs, atleast give some sort of thing to make it so that they aren't countered by as something as simple as researching advanced mops.

Funny you say that, every time I've seen a (s)laughter demon, shifts ended with 30 corpses still in its belly

**Comment by improvedname** [comment]
so your telling me a wizard can't remove people with shards since it absorbs the shades

**Comment by carlarctg** [comment]
slaughter demons are far too unkillable rn, they need some sort of way to limit their ability to play it safe. this is a good change

**Comment by Time-Green** [comment]
Maybe a better change would be to prevent demons from instantly re-entering blood? Or adding a small do-after when dragging corpses into blood?

Imo the fact that slaughter demons are more slippery than revenants is insane

**Comment by moocowswag** [comment]
Demons were balanced in the old days when lethal shotgun ammo was accessible.

**Comment by ATH1909** [comment]
Making slaughter demon jaunting be blocked by holy tiles (and take damage over time while on holy tiles, to help reduce GBJ situations) could also help with pinning them down and contain them? That might make chaplain spraying a firmer (and less fun) counter than cleaning, though...

**Review by MrMelbert — CHANGES_REQUESTED** [review]
I don't really have an intention to merge this, but whatever happens, make sure humans who get bloodcrawl through defeating demons aren't subject to the passive damage effect, I think that's lame

**Comment by Xackii** [comment]
> I don't really have an intention to merge this, but whatever happens, make sure humans who get bloodcrawl through defeating demons aren't subject to the passive damage effect, I think that's lame

let them suffer. 5 percent is 5 hp. It won't be difficult to heal. I don’t even understand what good comes from endless jaunt.

**Comment by MrMelbert** [comment]
"What good" is "it's fun"

**Comment by Xackii** [comment]
> "What good" is "it's fun"


Fun is good. For the last year, I haven’t seen anyone (killing a demon) using a demon’s heart. But usually when someone gets it, shuttle has already been called (because wizard) and you won’t be able to do anything particularly fun with acquired ability (except to show off). So I don't think that 5 hp every 20 seconds will change anything. Considering that it is not difficult for a crew member to inject himself with regenerating chemicals or stick his head out for a couple of seconds to tend his wounds

**Comment by MrMelbert** [comment]
If it's a show-off ability, then there's surely no issue with letting them go as much as they want

Regardless this is non-negotiable to me because I think it's harmless fun

**Comment by Holinka4ever** [comment]
> Slaughter Demon is currently super weak as it is, having relatively low melee-only damage for a lack of speed aside from ambushing and an incredibly easy weakness (just pull out space cleaner).
> 
> I've only seen slaughter demons somewhat recently do nothing but fail from experience because they can't hit shit whilst being countered by incredibly easily distributable cleaning supplies. If you're giving them nerfs, atleast give some sort of thing to make it so that they aren't countered by as something as simple as researching advanced mops.

Being an antagonist, that got summoned from other antagonist, getting more reliable tools rather than antagonist whom summoned them (infinite jaunt, instant and cooldown less roundremove by grabbing, full health by eating rats and dogs)
Named **slaughter demon** to hide in ghost form for whole shift, to act like **nightmare**, but not being em.
Its the most frustrating antag to fight with, nor wiz or nukes, nor even changeling with 4 hit armblade.
Though, its kinda salty, but its not only me frustraited from these semi-antag.

**Comment by Xackii** [comment]
carbon humans who ate the demon's heart do not receive damage while in jaunt.

**Comment by Metekillot** [comment]
> Maybe a better change would be to prevent demons from instantly re-entering blood? Or adding a small do-after when dragging corpses into blood?
> 
> Imo the fact that slaughter demons are more slippery than revenants is insane

You can hard counter a slaughter demon by modifying the timer on a space cleaner grenade to be instant. Wirecutter to open assembly, multi tool for advanced timer options.

**Comment by zoomachina** [comment]
> > Maybe a better change would be to prevent demons from instantly re-entering blood? Or adding a small do-after when dragging corpses into blood?
> > Imo the fact that slaughter demons are more slippery than revenants is insane
> 
> You can hard counter a slaughter demon by modifying the timer on a space cleaner grenade to be instant. Wirecutter to open assembly, multi tool for advanced timer options.

cleaner grenades are single use

**Comment by Metekillot** [comment]
> > > Maybe a better change would be to prevent demons from instantly re-entering blood? Or adding a small do-after when dragging corpses into blood?
> > > Imo the fact that slaughter demons are more slippery than revenants is insane
> > 
> > You can hard counter a slaughter demon by modifying the timer on a space cleaner grenade to be instant. Wirecutter to open assembly, multi tool for advanced timer options.
> 
> cleaner grenades are single use

... And infinitely refillable??

**Comment by Jacquerel** [comment]
This is an antagonist bought by another antagonist who is meant to end the round.
If you want to rework them then try doing something to making it so they need to keep doing their job (attacking people) in order to survive instead. Maybe if they go a couple minutes without drawing blood they will start dying (being pulled back to hell) until they do so.

It's better to encourage they're constantly in conflict, demons being cowards and hiding for ages is boring and worth targeting.
Demons just being hard to kill is fine.

**Comment by Holinka4ever** [comment]
> > > > Maybe a better change would be to prevent demons from instantly re-entering blood? Or adding a small do-after when dragging corpses into blood?
> > > > Imo the fact that slaughter demons are more slippery than revenants is insane
> > > 
> > > 
> > > You can hard counter a slaughter demon by modifying the timer on a space cleaner grenade to be instant. Wirecutter to open assembly, multi tool for advanced timer options.
> > 
> > 
> > cleaner grenades are single use
> 
> ... And infinitely refillable??

Everyone in shift has access to chemistry and they know recipe for cleaning.
Even though, while making it/googling recipe in wiki , slaughter demon can just beat you/chemist/break chemistry and boom, no more chemistry or way to heal up people.

**Comment by Holinka4ever** [comment]
> This is an antagonist bought by another antagonist who is meant to end the round.

Or mages with out enough own experience, just can buy demons, die in maints, spread hell on station and laugh in own supirioty (they didnt do anything particular on station)
 > If you want to rework them then try doing something to making it so they need to keep doing their job (attacking people) in order to survive instead. Maybe if they go a couple minutes without drawing blood they will start dying (being pulled back to hell) until they do so.

And then another maitainer would ask to change these into diffrent ideas, cause he finds something wrong and put "do not merge" label

> It's better to encourage they're constantly in conflict, demons being cowards and hiding for ages is boring and worth targeting.

But these pr why its made for, isnt it?

>Demons just being hard to kill is fine.
I understand if it because being tanky or dangerous to fight against
But infinite jaunt, without cooldown or radius of effect with option to eat animal corpses to heal yourself to full health.

I just saw a lot of prs, merged and closed, and ideas might good and interesting, but in other way really meh.
Thing is, someone told that they dont want to antags having free space to breath in breath out, and we have literally blood demon who have these ability. Heretic infinite jaunt is diffrent, its an unlockable skill, rather than demons.
Thats all what i have to say, its really weird to see that maintainers wish to save infinite jaunt for a semi-antag.

**Comment by zoomachina** [comment]
> It's better to encourage they're constantly in conflict, demons being cowards and hiding for ages is boring and worth targeting. Demons just being hard to kill is fine.

But this pr makes it so demon takes damage every 20 seconds in bloodcrawl (+no longer able to eat pets). Isn't this literally what you want to see?

**Comment by Jacquerel** [comment]
> > It's better to encourage they're constantly in conflict, demons being cowards and hiding for ages is boring and worth targeting. Demons just being hard to kill is fine.
> 
> But this pr makes it so demon takes damage every 20 seconds in bloodcrawl (+no longer able to eat pets). Isn't this literally what you want to see?

No because I don't care about how long they spend in bloodcrawl I care about how long they spend not fighting people, which we can target directly instead of indirectly.

**Comment by Xackii** [comment]
> This is an antagonist bought by another antagonist who is meant to end the round. 
Its just wizard purchase.
Guaridan deck also costs 2 points and it cannot kill the entire station; it requires the wizard to at least think.
> It's better to encourage they're constantly in conflict, demons being cowards and hiding for ages is boring and worth targeting. Demons just being hard to kill is fine.
Demons that are hard to kill are good. demons that cannot be killed are bad.

**Comment by jughu** [comment]
> > This is an antagonist bought by another antagonist who is meant to end the round.
> 
> Its just wizard purchase. Guaridan deck also costs 2 points and it cannot kill the entire station; it requires the wizard to at least think.
> 
> > It's better to encourage they're constantly in conflict, demons being cowards and hiding for ages is boring and worth targeting. Demons just being hard to kill is fine.
> 
> Demons that are hard to kill are good. Demons that cannot be killed are bad.

yeah but dex can also hold healing staff and click wizard whenever he gets low health and if wizard respects recharging he pretty much becomes unkillable im not gonna lie you gotta ttv bomb that man or throw a pipebomb at their face just massive damage in one go

**Comment by ExcessiveUseOfCobblestone** [comment]
are good demons really running into the issue of not killing people fast enough?

Are you saying theyre just waiting for a free kill which is why you want them to be in combat?

**Comment by Xackii** [comment]
> are good demons really running into the issue of not killing people fast enough?
> 
> Are you saying theyre just waiting for a free kill which is why you want them to be in combat?

As I already wrote above, most often demons live longer than their summoners, which may make sense, but when a wizard dies in the first 5 minutes and demon lives for more than an hour of the round, it’s more like a round with a demon than a round with a wizard. In my opinion, this should not be the case, demon must die (through his own fault or because he cannot find a target to kill). This is a ghost role(like guardian, syndicate borg, Ling spiders) that wizard buys for 2 points (or for 1 because hihihiha demon). The demon should spend more time trying to kill the crew rather than sitting in his jaunt and waiting for the right moment to attack.

**Comment by jughu** [comment]
just make the demon limited to 1 buy of each variant so wizards don't just spam it

**Comment by moocowswag** [comment]
Recommend holding off on this until https://github.com/tgstation/tgstation/pull/84024 as it will effect what the crew has access to in shorter rounds that will involve wizards and demons.

**Comment by ATH1909** [comment]
> yeah but dextrous holo can also hold healing staff and click wizard whenever he gets low health

didn't the staff of healing changes soft patch that out? or do dextrous holoparas owned by a wiz count as wizards for staff usage?

**Comment by carlarctg** [comment]
> > yeah but dextrous holo can also hold healing staff and click wizard whenever he gets low health
> 
> didn't the staff of healing changes soft patch that out? or do dextrous holoparas owned by a wiz count as wizards for staff usage?

they do count, saw a round with a dex para doing just that yesterday or so

**Comment by Xackii** [comment]
> Recommend holding off on this until #84024 as it will effect what the crew has access to in shorter rounds that will involve wizards and demons. Will likely make exotic ammunition more accessible.

will this bring shotguns back to their former glory

**Comment by ExcessiveUseOfCobblestone** [comment]
> As I already wrote above, most often demons live longer than their summoners, which may make sense, but when a wizard dies in the first 5 minutes and demon lives for more than an hour of the round, it’s more like a round with a demon than a round with a wizard

This is not a problem imo, both roles are being driven by players of varying skill. Left clicking ppl is a lot easier than getting a wacky assortment of spells you dont ever run into in any other part of the game either.

How the wizard chooses to offset his power is not a problem either personally, else we'd just say lets make blood demon a rare side antag spawn instead of making it specifically an opportunity cost with the wizards spellcasting ability.

> The demon should spend more time trying to kill the crew rather than sitting in his jaunt and waiting for the right moment to attack.

I see, that seems reasonable-ish. I guess the only problem is a good demon camping indefinitely is hard to discern from a bad demon who isnt finding any blood to work with, and in the latter im trying to understand why they "should" die in that case.

**Comment by Xackii** [comment]
> > As I already wrote above, most often demons live longer than their summoners, which may make sense, but when a wizard dies in the first 5 minutes and demon lives for more than an hour of the round, it’s more like a round with a demon than a round with a wizard
> 
> This is not a problem imo, both roles are being driven by players of varying skill. Left clicking ppl is a lot easier than getting a wacky assortment of spells you dont ever run into in any other part of the game either.
> 
> How the wizard chooses to offset his power is not a problem either personally, else we'd just say lets make blood demon a rare side antag spawn instead of making it specifically an opportunity cost with the wizards spellcasting ability.
> 
> > The demon should spend more time trying to kill the crew rather than sitting in his jaunt and waiting for the right moment to attack.
> 
> I see, that seems reasonable-ish. I guess the only problem is a good demon camping indefinitely is hard to discern from a bad demon who isnt finding any blood to work with, and in the latter im trying to understand why they "should" die in that case.

This is where dependence on the wizard comes in. At this point, the wizard is buying a demon just to ruin the round. Yes, perhaps someone does this to make gameplay easier, perhaps someone does this to let gosts to play, but most often it’s only because they have no idea what build to make for themselves. However, if you bought a demon, then the demon must be dependent on you. If you bought a demon, then you can at least make sure that there is blood all over the station: spill blood, cut corpses and pull across the station, and so on. If you bought a demon, then at least do some kind of gimmick around your purchase and not just: “I bought a demon to ruin round.” Again I give the example of Guardian: You can build some kind of build around him: give him a healing wand so that he can heal you, give him a Mjolner so that he can beat everyone, and so on. I understand that you can’t do any fun things with a demon (that’s why I don’t like a demon as a purchase from a wizard at all), but nevertheless a wizard can do at least something around the fact that the demon needs blood to play.

File: code/modules/spells/spell_types/jaunt/bloodcrawl.dm
Comment (Jacquerel) [inline code/modules/spells/spell_types/jaunt/bloodcrawl.dm:130]:
```suggestion
	lazy_demon.apply_damage(lazy_demon.maxHealth * 0.05, BRUTE)
```

File: code/modules/spells/spell_types/jaunt/bloodcrawl.dm
Comment (Jacquerel) [inline code/modules/spells/spell_types/jaunt/bloodcrawl.dm:111]: Make this a boolean check on the ability, not an iscarbon check.
Then make the heart grant a version with this check set to false.

Alternately, give the _demon_ a subtype of bloodcrawl which has this behaviour and the ability granted by the heart can be the base one (this might be better).

File: code/modules/spells/spell_types/jaunt/bloodcrawl.dm
Comment (Jacquerel) [inline code/modules/spells/spell_types/jaunt/bloodcrawl.dm:132]:
```suggestion
	to_chat(lazy_demon, span_warning("You feel your flesh dissolving into the sea of blood. You shouldn't stay in Blood Crawl for too long!"))
```

**Review by Jacquerel — CHANGES_REQUESTED** [review]
I decided my problems with this PR are minimal enough that we can move forward with it

**Comment by ShizCalev** [comment]
image
i'm in favor of closing. will leave it up to @Jacquerel

**Comment by Jacquerel** [comment]
I agree with the principle of "we should bully cowards who just hide in bloodcrawl instead of fighting people" and while it's not the exact solution I would apply it is fine enough
reaction count has never been a factor to me

File: code/modules/spells/spell_types/jaunt/bloodcrawl.dm
Comment (Xackii) [reply code/modules/spells/spell_types/jaunt/bloodcrawl.dm]: transferred everything from ``/datum/action/cooldown/spell/jaunt/bloodcrawl`` to ``/datum/action/cooldown/spell/jaunt/bloodcrawl/slaughter_demon`` sybtype. Demon heart give parent type so now anyone who use it don't deal with bloodcrawl damage.

File: code/modules/spells/spell_types/jaunt/bloodcrawl.dm
Comment (Jacquerel) [inline code/modules/spells/spell_types/jaunt/bloodcrawl.dm:278]: Add the `TIMER_STOPPABLE` flag to the `addtimer` call and call `deltimer` when the timer needs to be stopped.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Pull request merged**
